src/powerhouse/components/tree-view-item/tree-view-item.tsx

```diff
@@ -92,50 +79,55 @@ export const TreeViewItem: React.FC<TreeViewItemProps> = props => {
         <div {...divProps}>
             <div
                 role="button"
-                onClick={onClickItemHandler}
-                style={{
-                    paddingLeft: `${levelPadding + caretPadding}px`,
-
-                    ...containerButtonStyle,
-                }}
+                onClick={onClick}
+                style={containerButtonStyle}
                 className={twMerge(
-                    'flex flex-row w-full cursor-pointer select-none group/tree-item focus:outline-none',
+                    'flex flex-col w-full cursor-pointer select-none group/tree-item focus:outline-none relative max-h-[49px]',
                     containerButtonClassName,
                 )}
                 {...containerButtonProps}
             >
-                {children && (
-                    <img
-                        src={CaretIcon}
-                        className={twMerge(
-                            open && 'rotate-90',
-                            'transition ease delay-50 pointer-events-none',
-                        )}
-                    />
-                )}
-                {icon && (
-                    <img
-                        src={open ? expandedIcon || icon : icon}
-                        className="pointer-events-none"
-                    />
-                )}
-                {label && (
-                    <div className="ml-2 flex flex-1 overflow-hidden whitespace-nowrap relative">
-                        <span className="absolute right-0 w-12 h-full bg-gradient-to-r from-transparent to-inherit" />
-                        {label}
-                    </div>
-                )}
-                {optionsContent && (
-                    <div className="w-6 h-6 px-3 box-content hidden group-hover/tree-item:inline-block">
-                        {optionsContent}
-                    </div>
-                )}
-                {secondaryIcon && (
-                    <img
-                        src={secondaryIcon}
-                        className="flex self-end w-6 h-6 mx-3 group-hover/tree-item:hidden pointer-events-none"
-                    />
-                )}
+                <div className="absolute top-0 w-full">{topIndicator}</div>
```

Comment:
I would do `topIndicator && <div className="absolute top-0 w-full">{topIndicator}</div>` to avoid unnecessary divs.

Reply:
Added fix here: [00dd354](https://github.com/powerhouse-inc/design-system/pull/80/commits/00dd35439b91a035435c2066670a83248d8adde9)